tokio/src/sync/mpsc/bounded.rs

```diff
@@ -1002,6 +1075,86 @@ impl<T> Sender<T> {
         Ok(Permit { chan: &self.chan })
     }
 
+    /// Tries to acquire `n` slot in the channel without waiting for the slot to become
+    /// available.
+    ///
+    /// If the channel is full this function will return a [`TrySendError`], otherwise
+    /// A [`PermitIterator`] is returned to track the reserved capacity.
+    /// You can call this [`Iterator`] until it is exhausted to
+    /// get a [`Permit`] and then call [`Permit::send`]. This function is similar to
+    /// [`reserve_many`] except it does not await for the slot to become available.
```

Comment:
A few notes on the wording here:

 * We should explain the normal case first, so start I would prefer to start the paragraph by talking about the case where you get a `PermitIterator`.
 * You say that it returns a `TrySendError` when the channel is full. That is not wrong, but imprecise. You should say which _kind_ of `TrySendError` it returns. (That is, `TrySendError::Full`.)
 * There are other cases than "channel is full" where you get the `TrySendError::Full` error. It also happens if there are fewer than `n` permits available, even if the number of available permits is non-zero.
 * You should mention that this can also fail with a `TrySendError::Closed` error.
 * If you make the other function panic on `n > self.max_capacity()`, then you should explicitly say that this method *doesn't* panic in that case, since we can just handle it by returning `TrySendError::Full`.